--- a/doc/Muttrc.head
+++ b/doc/Muttrc.head
@@ -18,7 +18,7 @@
 # simulate the old url menu
 macro index,pager,attach,compose \cb "\
 <enter-command> set my_pipe_decode=\$pipe_decode pipe_decode<Enter>\
-<pipe-message> urlview<Enter>\
+<pipe-message> ([ -x /usr/bin/urlview ] && urlview || echo \"Please install the urlview package\")<Enter>\
 <enter-command> set pipe_decode=\$my_pipe_decode; unset my_pipe_decode<Enter>" \
 "call urlview to extract URLs out of a message"
 
